rusk-wallet/src/bin/command.rs

```diff
@@ -808,54 +808,58 @@ impl fmt::Display for RunResult {
         use RunResult::*;
         match self {
             PhoenixBalance(balance, _) => {
+                let total = Dusk::from(balance.value);
+                let spendable = Dusk::from(balance.spendable);
                 write!(
                     f,
-                    "> Total Phoenix balance is: {} DUSK\n> Maximum spendable per TX is: {} DUSK",
-                    Dusk::from(balance.value),
-                    Dusk::from(balance.spendable)
+                    "> Total Phoenix balance is: {total} DUSK\n\
+                     > Maximum spendable per TX is: {spendable} DUSK",
                 )
             }
             MoonlightBalance(balance) => {
-                write!(f, "> Total Moonlight balance is: {} DUSK", balance)
+                write!(f, "> Total Moonlight balance is: {balance} DUSK")
             }
             Address(addr) => {
-                write!(f, "> {}", addr)
+                write!(f, "> {addr}")
             }
             Addresses(addrs) => {
                 let str_addrs = addrs
                     .iter()
-                    .map(|a| format!("{}", a))
+                    .map(|a| format!("{a}"))
                     .collect::<Vec<String>>()
-                    .join("\n>");
+                    .join("\n> ");
                 write!(f, "> {}", str_addrs)
             }
             Tx(hash) => {
                 let hash = hex::encode(hash.to_bytes());
                 write!(f, "> Transaction sent: {hash}",)
             }
             StakeInfo(data, _) => {
-                let stake_str = match data.amount {
-                    Some(amt) => format!(
-                        "Current stake amount is: {} DUSK\n> Stake eligibility from block #{} (Epoch {})",
-                        Dusk::from(amt.value),
-                        amt.eligibility,
-                        amt.eligibility / EPOCH
-                    ),
-                    None => "No active stake found for this key".to_string(),
-                };
-                write!(
-                    f,
-                    "> {}\n> Accumulated reward is: {} DUSK",
-                    stake_str,
-                    Dusk::from(data.reward)
-                )
+                match data.amount {
+                    Some(amt) => {
+                        let amount = Dusk::from(amt.value);
```

Comment:
we should have the stake_info function in the wallet impl and kt should return all this info so it can also be called as a function as a library

Reply:
All those infos are already in the execution-core stakeData structure